lte/cloud/go/services/subscriberdb/storage/ip_lookup_test.go

```diff
@@ -113,13 +113,20 @@ func TestIPLookup(t *testing.T) {
 			{Ip: "ipB", Imsi: "IMSI2", Apn: "apn1"},
 			{Ip: "ipC", Imsi: "IMSI2", Apn: "apn2"},
 		}
-		assert.Equal(t, want, got)
+		assertEqualArrays(t, want, got)
 
 		got, err = s.GetIPs("n1", []string{"ipA", "ipB", "ipC", "ipZ"})
 		assert.NoError(t, err)
 		want = []*protos.IPMapping{
 			{Ip: "ipZ", Imsi: "IMSI0", Apn: "apn0"},
 		}
-		assert.Equal(t, want, got)
+		assertEqualArrays(t, want, got)
 	})
 }
+
+func assertEqualArrays(t *testing.T, expected []*protos.IPMapping, actual []*protos.IPMapping) {
```

Comment:
For my understanding: why is this necessary over assert.Equal?

Reply:
The root problem here is the new proto implementation of messages. Messages now contain other (unexported) meta fields whose values make the comparison fail.
To avoid that one can either use .String() (not recommended by protobuf due to unstable representation) or proto.Equal(...).

Just for "illustration":
```
ip_lookup_test.go:57: 
            	Error Trace:	ip_lookup_test.go:57
            	Error:      	Not equal: 
            	            	expected: []*protos.IPMapping{(*protos.IPMapping)(0xc0001151a0), (*protos.IPMapping)(0xc000115200)}
            	            	actual  : []*protos.IPMapping{(*protos.IPMapping)(0xc000114ea0), (*protos.IPMapping)(0xc000114f60)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -9,3 +9,1337 @@
            	            	    },
            	            	-   atomicMessageInfo: (*impl.MessageInfo)(<nil>)
            	            	+   atomicMessageInfo: (*impl.MessageInfo)({
            	            	+    GoReflectType: (*reflect.rtype)({
```